pkg/core/object/metadata.go

```diff
@@ -0,0 +1,128 @@
+package object
+
+import (
+	"bytes"
+	"math/big"
+	"slices"
+	"strings"
+
+	"github.com/nspcc-dev/neofs-sdk-go/client"
+)
+
+// TODO: docs.
+func MergeSearchResults(lim uint16, withAttr bool, sets [][]client.SearchResultItem, mores []bool) ([]client.SearchResultItem, bool) {
+	if lim == 0 || len(sets) == 0 {
+		return nil, false
+	}
+	if len(sets) == 1 {
+		n := min(uint16(len(sets[0])), lim)
+		return sets[0][:n], n < lim || slices.Contains(mores, true)
+	}
+	lim = calcMaxUniqueSearchResults(lim, sets)
+	res := make([]client.SearchResultItem, 0, lim)
+	var more bool
+	var minInt *big.Int
+	for minInd := -1; ; minInd, minInt = -1, nil {
```

Comment:
I think it'd be much easier with https://pkg.go.dev/container/heap, you only need `len(sets)` of elements to store there, push all of them into heap, then pick the lowest and push a new one from the same queue.

Reply:
i'd try this withinin a separate issue if u dont mind